File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/GameClient.java
package uk.co.ourfriendirony.medianotifier.clients;

import java.io.IOException;
import java.util.List;

import uk.co.ourfriendirony.medianotifier.clients.rawg.RAWGClient;
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem;

public class GameClient implements Client {

    @Override
    public List<MediaItem> searchMediaItem(String name) throws IOException {
        return new RAWGClient().queryGame(name);
    }

    @Override
    public MediaItem getMediaItem(String id) throws IOException {
        return new RAWGClient().getGame(id);
    }
}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/clients/GameClient.kt
package uk.co.ourfriendirony.medianotifier.clients

import uk.co.ourfriendirony.medianotifier.clients.rawg.RAWGClient
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem
import java.io.IOException

class GameClient : Client {
    @Throws(IOException::class)
    override fun searchMediaItem(name: String?): List<MediaItem?> {
        return RAWGClient().queryGame(name)
    }

    @Throws(IOException::class)
    override fun getMediaItem(id: String?): MediaItem {
        return RAWGClient().getGame(id)
    }
}